src/runtime/pickle_runner.ts

```diff
@@ -366,6 +381,26 @@ export default class PickleRunner {
     return await this.invokeStep(null, hookDefinition, hookParameter)
   }
 
+  async runStepHook(
+    stepHookDefinition: TestStepHookDefinition,
+    stepResult: messages.TestStepFinished.ITestStepResult
+  ): Promise<messages.TestStepFinished.ITestStepResult> {
+    if (this.isSkippingSteps()) {
+      return messages.TestStepFinished.TestStepResult.fromObject({
+        status: Status.SKIPPED,
+      })
+    }
```

Comment:
This isSkippingSteps check should go away. We'll run the step hook if we are running the step. This function could probably be removed given the other simplifications and merge this into runStepHooks